### ISSUE TITLE
Add icons for mimetypes

### ISSUE THREAD

**Issue by jusw85** [garglk/garglk#461]
Add icons to mimetypes for file manager integration.

e.g. in interactive-fiction.xml:
```
  <mime-type type="application/x-adrift">
    <comment>Adrift game data</comment>
    <magic priority="50">
      <match type="string" offset="0" value="\x3C\x42\x3F\xC9\x6A\x87\xC2\xCF"/>
    </magic>
    <glob pattern="*.taf"/>
    <icon name="application-x-gargoyle"/>
  </mime-type>
```

As per [FreeDesktop Icon Naming Specification](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-0.8.html#mimetypes), I would recommend renaming `gargoyle-docu2.png` to `application-x-gargoyle.png`

**Comment by cspiegel** [comment]
I agree that adding an icon has utility; I'm a bit unsure how this ought to be done, though.

Ideally `interactive-fiction.xml` wouldn't be handled by Gargoyle at all.  It's more generic than that (and as a note, it overlaps with a MIME package provided by QTads, which naturally takes over the TADS file types).  I think the best thing would be to get this integrated into `shared-mime-info` although I don't know the process to do that; do you have any ideas?  The real problem, I think, is that this hasn't been standardized at all, so we don't even have real MIME types: I just made ones up so _something_ could be assigned in the `.desktop` file.

I'm fine updating this for now, but ultimately it'd be best to integrate it elsewhere.  I just haven't put any effort into doing so.

**Comment by jusw85** [comment]
After further investigation, I realise this change isn't needed - the [default icon name](https://specifications.freedesktop.org/shared-mime-info-spec/shared-mime-info-spec-0.21.html#idm46070616650832) can be defined by replacing '/' with '-'

```
Generally the icon used for a mimetype is created based on the mime-type by mapping "/" characters to "-", but users can override this by using the icon element to customize the icon for a particular mimetype.
```

I can install the icon as `application-x-adrift.png`, `application-x-blorb.png`, etc. in `/usr/share/icons/hicolor/32x32/mimetypes`. After updating the icon cache, it should be picked up by competent desktops / file managers (it does for mine).

Regarding upstream integration, it can be added by creating an enhancement request in the [shared-mime-info project](https://gitlab.freedesktop.org/xdg/shared-mime-info/-/issues), which is then added to [this file](https://gitlab.freedesktop.org/xdg/shared-mime-info/-/blob/master/data/freedesktop.org.xml.in).

The issue as you mentioned is, are these definitions standardised/definitive? Are the magic numbers / globs likely to change in the future? Would this agree with current/future IF software e.g. QTads? If there are no issues, I think it would be a good idea to open an issue push this upstream in the shared mime database.

**Comment by cspiegel** [comment]
> Regarding upstream integration, it can be added by creating an enhancement request in the shared-mime-info project, which is then added to this file.

That would be the best approach without question, but...

> The issue as you mentioned is, are these definitions standardised/definitive? Are the magic numbers / globs likely to change in the future? Would this agree with current/future IF software e.g. QTads?

The globs shouldn't ever change: most of these are dead formats, and those that are alive aren't moving targets in any way.

For the same reason the magic numbers shouldn't be changing either.  Some are "obviously" correct, although others I probably deduced from looking at hexdumps/source code, so there may be room for improvement.

There's no argument with QTads at all: the definitions are identical (but for the comment, which I'd sync with QTads).

I'd completely forgotten that I did a half-hearted push for this several years ago (https://intfiction.org/t/mime-types-for-various-game-types/11322).  So now's a good time as any to try to upstream these.  A possible sticking point is the MIME type, I think, but there have been several of them used in the wild (e.g. at the IF Archive) for years, so at least it's not completely out of the blue.

@realnc Are you willing to have your TADS and Hugo MIME definitions integrated into an attempt to add IF MIME types to `shared-mime-info`?

**Comment by realnc** [comment]
> @realnc Are you willing to have your TADS and Hugo MIME definitions integrated into an attempt to add IF MIME types to `shared-mime-info`?

Sure. Although it's best to still ship these XML files with interpreters. Not everybody will have an up-to-date version of shared-mime-data.

**Comment by cspiegel** [comment]
> > @realnc Are you willing to have your TADS and Hugo MIME definitions integrated into an attempt to add IF MIME types to `shared-mime-info`?
> 
> Sure. Although it's best to still ship these XML files with interpreters. Not everybody will have an up-to-date version of shared-mime-data.

That makes sense; there's certainly no harm in having them as widely available as possible.  Thank you!
